client/services/SolutionHandlerService.js

```diff
@@ -45,19 +45,28 @@ tie.factory('SolutionHandlerService', [
           // Otherwise, the code doesn't have any obvious syntax errors.
           // Generate a CodeSubmission object that wraps the student's code
           // into a class and appends some test code, then run the whole thing.
+          var correctnessTests = [];
+          var buggyOutputTests = [];
+          var performanceTests = [];
+          for (var i = 0; i < tasks.length; i++) {
+            correctnessTests.push(tasks[i].getCorrectnessTests());
+            buggyOutputTests.push(tasks[i].getBuggyOutputTests());
+            performanceTests.push(tasks[i].getPerformanceTests());
+          }
+          var lastTask = tasks[tasks.length - 1];
           var codeSubmission = CodeSubmissionObjectFactory.create(
             studentCode.trim());
           CodePreprocessorDispatcherService.preprocess(
             language, codeSubmission, auxiliaryCode,
-            task.getInputFunctionName(), task.getMainFunctionName(),
-            task.getOutputFunctionName(), task.getCorrectnessTests(),
-            task.getBuggyOutputTests(), task.getPerformanceTests());
+            lastTask.getInputFunctionName(), lastTask.getMainFunctionName(),
```

Comment:
It still works. Our impl just uses the code that the student submits for the current task and test it on all test cases.